backend/src/contexts/users/actions/SignInAction.test.ts

```diff
@@ -6,38 +6,48 @@ import { UserRepositoryInMemory } from '../infrastructure/persistence/repositori
 import { SignInAction } from './SignInAction';
 
 describe('signing in', () => {
+  test('cant sign in because the username does not exist', async () => {
+    const userRepository = new UserRepositoryInMemory();
+    const signInAction = new SignInAction(userRepository);
 
-  test('cant sign in because the user does not exist', async () => {
+    await expect(signInAction.execute('password', 'username')).rejects.toThrow(
+      UserNotFoundException,
+    );
+  });
+
+  test('cant sign in because the email does not exist', async () => {
     const userRepository = new UserRepositoryInMemory();
     const signInAction = new SignInAction(userRepository);
 
-    await expect(signInAction.execute('username', 'password')).rejects.toThrow(
+    await expect(signInAction.execute('password', 'email')).rejects.toThrow(
       UserNotFoundException,
     );
   });
 
   test('cant sign in because the password is wrong', async () => {
     const userRepository = new UserRepositoryInMemory();
     const username = 'username';
+    const email = 'email';
     const password = 'password';
-    userRepository.storeUser(new User('email', username, password));
+    userRepository.storeUser(new User(email, username, password));
     const signInAction = new SignInAction(userRepository);
 
     await expect(
-      signInAction.execute(username, 'wrong password'),
+      signInAction.execute('wrong password', username, email),
     ).rejects.toThrow(WrongPasswordException);
   });
 
   test('can sign in', async () => {
```

Comment:
This test it's can sign in by username, but we need another about can sign in by email

Reply:
Done